nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/DeleteHBaseRow.java

```diff
@@ -0,0 +1,213 @@
+        List<FlowFile> flowFiles = session.get(flowFileCount);
+
+        if (flowFiles != null && flowFiles.size() > 0) {
+            for (int index = 0; index < flowFiles.size(); index++) {
+                FlowFile flowFile = flowFiles.get(index);
+                final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+                try {
+                    if (location.equals(ROW_ID_BODY.getValue())) {
+                        flowFile = doDeleteFromBody(flowFile, context, session, tableName, batchSize, charset);
+                        if (flowFile.getAttribute(RESTART_INDEX) != null) {
+                            session.transfer(flowFile, REL_FAILURE);
+                        } else {
+                            final String transitUrl = clientService.toTransitUri(tableName, flowFile.getAttribute(ROWKEY_END));
+                            session.transfer(flowFile, REL_SUCCESS);
+                            session.getProvenanceReporter().invokeRemoteProcess(flowFile, transitUrl);
+                        }
+                    } else {
+                        String transitUrl = doDeleteFromAttribute(flowFile, context, tableName, charset);
+                        session.transfer(flowFile, REL_SUCCESS);
+                        session.getProvenanceReporter().invokeRemoteProcess(flowFile, transitUrl);
+                    }
+                } catch (Exception ex) {
+                    getLogger().error(ex.getMessage(), ex);
+                    session.transfer(flowFile, REL_FAILURE);
+                }
+            }
+        }
+    }
+
+    private String doDeleteFromAttribute(FlowFile flowFile, ProcessContext context, String tableName, String charset) throws Exception {
+        String rowKey = context.getProperty(ROW_ID).evaluateAttributeExpressions(flowFile).getValue();
+        clientService.delete(tableName, rowKey.getBytes(charset));
+
+        return clientService.toTransitUri(tableName, rowKey);
+    }
+
+    private FlowFile doDeleteFromBody(FlowFile flowFile, ProcessContext context, ProcessSession session, String tableName, int batchSize, String charset) throws Exception {
+        String keySeparator = context.getProperty(KEY_SEPARATOR).evaluateAttributeExpressions(flowFile).getValue();
+        final String restartIndex = flowFile.getAttribute(RESTART_INDEX);
+
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        session.exportTo(flowFile, out);
+        out.close();
+
+        String data = new String(out.toByteArray(), charset);
+
+        int restartFrom = -1;
+        if (restartIndex != null) {
+            restartFrom = Integer.parseInt(restartIndex);
+        }
+
+        String first = null, last = null;
+
+        List<byte[]> batch = new ArrayList<>();
+        if (data != null && data.length() > 0) {
+            String[] parts = data.split(keySeparator);
```

Comment:
The usage says "Delete HBase records individually or in batches. The input can be a single row ID in the body, one ID per line, row IDs separated by commas or a combination of the two."
However, the code does not treat line separator. Should we update the doc or code?

Reply:
I updated the documentation because the code is the way to go here since it should be a configurable separator.